packages/flutter/lib/src/services/raw_keyboard.dart

```diff
@@ -534,14 +574,28 @@ class RawKeyboard {
     // Make sure that the modifiers reflect reality, in case a modifier key was
     // pressed/released while the app didn't have focus.
     _synchronizeModifiers(event);
-    if (_listeners.isEmpty) {
-      return;
-    }
+    // Send the event to passive listeners.
     for (final ValueChanged<RawKeyEvent> listener in List<ValueChanged<RawKeyEvent>>.from(_listeners)) {
       if (_listeners.contains(listener)) {
         listener(event);
       }
     }
+    // Send the key event to the keyEventHandler, and send the appropriate
+    // response to the platform so that it can resolve the event's handling.
+    _sendKeyEventResponse(event, handled: keyEventHandler?.call(event) ?? false);
```

Comment:
Should we assert here that if the keyEventHandler was provided it returned true/false (and not null)? Assuming that returning null from keyEventHandler is illegal.

Reply:
If null is illegal, let's document that on RawKeyEventHandler as well.

Reply:
Done.